### PR TITLE
Added support for swagger-ui.

### PR DESCRIPTION
**Pull request by jfly**
See #1078.

@ronaldmansveld could you take a look? I realize this is pretty railsy, but hopefully you can give this a spin and tell me how this looks?

http://localhost:3000/api/swaggers/v0 will run swagger-ui on our public/api/v0/swagger.yaml file.

### CODE REVIEW

File: WcaOnRails/public/api/v0/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+paths:
+  /competitions:
+    get:
+      summary: Competition List
+      description: |
+        This grabs all future competitions
+      parameters:
+        - name: delegate
+          in: query
+          description: Grabs competitions where the delegate matches the this WCA ID
```
Comment (jonatanklosko) [inline WcaOnRails/public/api/v0/swagger.yaml]: `the this` =) ?

File: WcaOnRails/public/api/v0/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            items:
+              $ref: '#/definitions/Competition'
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+  /me:
+    get:
+      summary: User Profile
+      description: The User Profile endpoint returns information about the Uber user that has authorized with the application.
```
Comment (jonatanklosko) [inline WcaOnRails/public/api/v0/swagger.yaml]: Nit: `Uber` -> `WCA`

File: WcaOnRails/public/api/v0/swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    properties:
+      class:
+        type: string
+        description: String set as "User"
+      id:
+        type: integer
+        description: WCA User ID
+      name:
+        type: string
+        description: first and last name of the user.
```
Comment (jonatanklosko) [inline WcaOnRails/public/api/v0/swagger.yaml]: I guess we should follow a convention for this, i.e. capitalized & ending with a dot, right?

File: WcaOnRails/public/api/v0/swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      gender:
+        type: string
+        enum: ['m', 'f']
```
Comment (jonatanklosko) [inline WcaOnRails/public/api/v0/swagger.yaml]: Shouldn't it be this way?
```yaml
enum:
- m
- f
```

**Comment by jfly** [comment]
@jonatanklosko, thanks for the comments! As you correctly predicted, the yaml file is really just there for show, I just copied it from something @coder13 started a few weeks ago.

Hopefully @ronaldmansveld can provide some insight into how we should get started working on swagger. I could see us implementing it, but just not linking to it anywhere until we feel it's at a good enough spot where people could benefit from seeing it.

**Comment by jfly** [comment]
I'm going to close this in favor of sticking with an ad-hoc, undocumented API for the next year or two. After that point in time, we'll look at all the APIs we have, and start work on a `v1` that is better thought out and actually documented.

We could certainly do something different if someone else (not me) is motivated to take the lead on this.

**Pull request closed without merging**